### PR TITLE
 Clean up memory after ReadGTACmdArgument/GetServerPort calls

### PR DESCRIPTION
**Pull request by marcelgerber**
As noted in #45. @agrippa1994 

Actual fix is in f31ac13c0fcce26a9bdd0b28b30bd2b818b31f69, and then there's also some general cleanup in f3c52c2c.

### CODE REVIEW

File: src/Open-SAMP-API/Client/SAMPFunctions.cpp [hunk 1]
```diff
@@ -2,48 +2,56 @@
...
-	ZeroMemory(portStr, 5);
-	if (ReadGTACmdArgument("-p", portStr, 5)) {
+	auto *portStr = new char[6];
+	ZeroMemory(portStr, 6);
+
+	BOOST_SCOPE_EXIT_ALL(portStr) {
+		delete[] portStr;
+	};
+
+	if (ReadGTACmdArgument("-p", portStr, 6)) {
```
Comment (marcelgerber) [inline src/Open-SAMP-API/Client/SAMPFunctions.cpp:52]: Didn't know whether we should use 5 or 6.
Guess with 6, we can still end up with a non-terminated string?

Comment (marcelgerber) [reply src/Open-SAMP-API/Client/SAMPFunctions.cpp:52]: Then again, we do the same thing down in `GetPlayerId` where we allocate a `char[25]` and call `GetPlayerName` with `max_len=25`.
There, too, we should never exceed the SAMP limit of 20 chars.

**Comment by marcelgerber** [comment]
@agrippa1994 Any thoughts?

**Comment by agrippa1994** [comment]
PR looks good! In my opinion it does not matter if we allocate 6, 7 or even 128 bytes on the heap since resources are not a problem nowadays. I would stick to more memory, just in case.

Unfortunately I can not browse the code in a convinient way at the moment but I assume that GetPlayerName handles memory that exceeds the limits properly. At least it should do it :)

**Pull request merged**